daemon/api_notices.go

```diff
@@ -124,5 +192,19 @@ func getNotice(c *Command, r *http.Request, user *auth.UserState) Response {
 	if notice == nil {
 		return NotFound("cannot find notice with id %q", noticeID)
 	}
+	if !noticeViewableByUser(notice, requestUID) {
+		return Forbidden("not allowed to access notice with id %q", noticeID)
+	}
 	return SyncResponse(notice)
 }
+
+func noticeViewableByUser(notice *state.Notice, requestUID uint32) bool {
+	userID, isSet := notice.UserID()
+	if !isSet {
+		return true
+	}
+	if requestUID == 0 {
```

Comment:
probably a comment mentiong root can view anything would be useful, as we don't have a helper involved here like pebble does